test/api/profileTypes.test.js

```diff
@@ -1,33 +1,318 @@
 
+const correctRequest = {
+  uri: '/api/profileTypes',
+  method: METHODS.GET,
+  mockResponse: PROFILE_TYPES_OK_PAGE_1,
+  useAuthentication: true,
+};
+
+
 jest.unmock('api/profileTypes');
-jest.useFakeTimers();
+jest.mock('@entando/apimanager', () => ({
+  makeRequest: jest.fn(() => new Promise(resolve => resolve({}))),
+  METHODS: {
+    GET: 'GET', POST: 'POST', PUT: 'PUT', DELETE: 'DELETE',
+  },
+}));
+
+describe('api/postProfileType', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+  it('returns a promise', () => {
+    expect(postProfileType()).toBeInstanceOf(Promise);
+  });
+
+  it('if successful, returns a mock ok response', () => {
+    postProfileType(PROFILE_TYPES);
+    expect(makeRequest).toHaveBeenCalledWith({
+      ...correctRequest,
+      method: 'POST',
+      mockResponse: PROFILE_TYPES,
+      body: PROFILE_TYPES,
+    });
+  });
+});
+
+describe('api/putProfileType', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+  it('returns a promise', () => {
+    expect(putProfileType(PROFILE_TYPES)).toBeInstanceOf(Promise);
+  });
+
+  it('if successful, returns a mock ok response', () => {
+    putProfileType(PROFILE_TYPES);
+    expect(makeRequest).toHaveBeenCalledWith({
+      ...correctRequest,
+      uri: '/api/profileTypes/AAA',
+      method: 'PUT',
+      mockResponse: PROFILE_TYPES,
+      body: PROFILE_TYPES,
+    });
+  });
+});
+
+describe('api/getProfileType', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+  it('returns a promise', () => {
+    expect(getProfileType('AAA')).toBeInstanceOf(Promise);
+  });
+
+  it('if successful, returns a mock ok response', () => {
+    getProfileType('AAA');
+    expect(makeRequest).toHaveBeenCalledWith({
+      ...correctRequest,
+      uri: '/api/profileTypes/AAA',
+      method: 'GET',
+      mockResponse: PROFILE_TYPES,
+    });
+  });
+});
+
+
+describe('api/deleteProfileType', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+  it('returns a promise', () => {
+    expect(deleteProfileType('AAA')).toBeInstanceOf(Promise);
+  });
+
+  it('if successful, returns a mock ok response', () => {
+    deleteProfileType('AAA');
+    expect(makeRequest).toHaveBeenCalledWith({
+      ...correctRequest,
+      uri: '/api/profileTypes/AAA',
+      method: 'DELETE',
+      mockResponse: PROFILE_TYPES_DELETE_OK,
+      body: 'AAA',
+    });
+  });
+});
+
+describe('/api/profileTypes/', () => {
+  describe('getAttributeFromProfileType', () => {
+    beforeEach(() => {
+      jest.clearAllMocks();
+    });
+    it('returns a promise', () => {
+      expect(getAttributeFromProfileType()).toBeInstanceOf(Promise);
+    });
 
-describe('api/profileTypes', () => {
-  afterEach(() => {
-    jest.runOnlyPendingTimers();
+    it('if successful, returns a mock ok response', () => {
+      getAttributeFromProfileType('AAA', 'code');
+      expect(makeRequest).toHaveBeenCalledWith({
+        ...correctRequest,
```

Comment:
Same here, why copying the request and then override almost everything?
this applies also to all of the following cases